kdc/fast.c

```diff
@@ -799,8 +799,10 @@ _kdc_fast_strengthen_reply_key(astgs_request_t r)
 
 	ret = krb5_generate_random_keyblock(r->context, r->reply_key.keytype,
 					    &r->strengthen_key);
-	if (ret)
-	    krb5_abortx(r->context, "random generator fail");
+	if (ret) {
+	    kdc_log(r->context, r->config, 0, "random generator fail");
+	    return ret;
+	}
```

Comment:
Should we say why it failed - given the reason we are not abort()ing is that it can fail for reasons unrelated to the RNG?